pymeasure/instruments/hp/hp3478A.py

```diff
@@ -463,3 +463,88 @@ def trigger(self):
     def trigger(self, value):
         trig_set = self.TRIGGERS[strict_discrete_set(value, self.TRIGGERS)]
         self.write(trig_set)
+
+    @property
+    def calibration_data(self):
+        """Return the full calibration data as an array of 256 values between 0 and 15.
+
+        The calibration data of an HP 3478A is stored in a 256x4 SRAM that is
+        permanently powered by a 3v Lithium battery. When the battery runs
+        out, the calibration data is lost, and recalibration is required.
+
+        This property fetches the calibration data so that it can be backed up.
+
+        """
+        cal_data = []
+        for addr in range(0, 256):
+            # To fetch one nibble: 'W<address>', where address is a raw 8-bit number.
+            cmd = bytes([ord('W'), addr])
+            self.write_bytes(cmd)
+            rvalue = self.read_bytes(1)[0]
+            if rvalue < 64 or rvalue >= 80:
+                raise Exception("calibration nibble out of range")
+            cal_data.append(rvalue-64)
+
+        return cal_data
+
+    @calibration_data.setter
+    def calibration_data(self, cal_data):
+        """Write the full calibration data to the instrument
+
+        Expects an array of 256 values between 0 and 15.
+
+        The CAL ENABLE switch at the front of the instrument must be set to ON,
+        otherwise an exception is raised.
+
+        IMPORTANT: changing the calibration data results in permanent loss of
+        the previous data. Use with care!
+
+        """
+
+        if not(self.calibration_enabled):
+            raise Exception("CAL ENABLE switch not set to ON")
+        if len(cal_data) != 256:
+            raise Exception("cal_data must contain 256 values")
+
```

Comment:
Should you use the checksum verification from below on `cal_data` before starting the write?

Reply:
I also thought about that.
Its another check preventing erroneous change and thus loss of data.

Reply:
I'll make the check enabled by default, with a function parameter to override: I definitely want to be able to write corrupt data to check that the write function is really working.

Reply:
Can't you test the write function with a working calibration data set? Maybe using two different sets to verify, that it changes?